chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,350 @@
+from chainer.backends import cuda
+import chainer.functions as F
```

Comment:
To avoid cyclic dependency in executing `import chainer`, do not use alias names such as `chainer.functions.expand_dims`. Could you use full path such as `chainer.functions.array.expand_dims.expand_dims`? Implementation of VAE losses [code](https://github.com/chainer/chainer/blob/master/chainer/functions/loss/vae.py) might be helpful.